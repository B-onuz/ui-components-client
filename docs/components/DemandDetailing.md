# DemandDetailing (Component)

Componente de detalhamento de demana

## Como utilizar

```js
import DemandDetailing from 'ui-components/lib/components/DemandDetailing'
```

```jsx
import React from 'react'
import DemandDetailing from './DemandDetailing'
import { Box, Flex } from 'reflexbox'
import Button from '../Button'
import Typography from '../Typography'

export default {
  title: 'DemandDetailing',
  component: DemandDetailing,
}

const data = [
  {
    name: 'Advogado Bonuz',
    date: '22/08/2020 08:45',
    text:
      'Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum ',
  },
  {
    name: 'Cliente',
    date: '22/08/2020 08:45',
    text:
      'Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum ',
  },
  {
    name: 'Advogado Bonuz',
    date: '22/08/2020 08:45',
    text:
      'Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum ',
    attachments: [
      {
        url: 'https://lorempixel.com/50/50',
        label: 'contrato_nda_v1.pdf',
      },
      {
        url: 'https://lorempixel.com/50/50',
        label: 'contrato_nda_v2.pdf',
      },
    ],
  },
]

export const DemandDetailingExample = () => <DemandDetailing reportTitle="Conclusão" reportKind="Histórico" detailingData={data} />
```
