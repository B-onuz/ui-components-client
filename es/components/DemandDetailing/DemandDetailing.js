"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _styledSystem = require("styled-system");

var _reflexbox = require("reflexbox");

var _Fieldset = _interopRequireDefault(require("../Fieldset"));

var _UserAvatar = _interopRequireDefault(require("../UserAvatar"));

var _Typography = _interopRequireDefault(require("../Typography"));

var _Legend = _interopRequireDefault(require("../Legend"));

var _Button = _interopRequireDefault(require("../Button"));

var _utils = require("../../utils");

var _this = void 0,
    _jsxFileName = "/Users/master/projects/bonuz/ui-components/src/components/DemandDetailing/DemandDetailing.js";

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

function _templateObject5() {
  var data = _taggedTemplateLiteralLoose(["\n  display: flex;\n  flex-direction: column;\n  padding-left: 52px;\n\n  @media only screen and (max-width: 500px) {\n    padding-left: 30px;\n  }\n"]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4() {
  var data = _taggedTemplateLiteralLoose(["\n  margin-bottom: 28px;\n  margin-left: 38px;\n  &.mainTopic {\n    margin-left: -20px;\n  }\n  &.lastTopic {\n    margin-bottom: 0;\n  }\n\n  @media only screen and (max-width: 500px) {\n    margin-left: 10px;\n  }\n"]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = _taggedTemplateLiteralLoose(["\n  border-bottom: 1px solid #dededf;\n  margin-bottom: 28px;\n"]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = _taggedTemplateLiteralLoose(["\n  ", "\n  display: flex;\n  flex-direction: column;\n  border-left: 1px solid #dededf;\n  margin-bottom: 40px;\n"]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteralLoose(["\n  ", "\n  ", "\n  legend {\n    font-size: 21px;\n  }\n  @media only screen and (max-width: 500px) {\n    margin-left: 20px;\n    padding-left: 0px;\n  }\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

function _taggedTemplateLiteralLoose(strings, raw) { if (!raw) { raw = strings.slice(0); } strings.raw = raw; return strings; }

var StyledDemandDetailing = _styledComponents["default"].div(_templateObject(), _styledSystem.color, _styledSystem.space);

var ContentWrapper = _styledComponents["default"].div(_templateObject2(), _styledSystem.color);

var ReportBodyTitle = _styledComponents["default"].h3(_templateObject3());

var TopicItemWrapper = _styledComponents["default"].div(_templateObject4());

var TopicItem = _styledComponents["default"].div(_templateObject5());

var getTopicClass = function getTopicClass(value, parameter) {
  if (value === 0) {
    return 'mainTopic';
  } else if (value === parameter - 1) {
    return 'lastTopic';
  } else {
    return null;
  }
};

var DemandDetailing = function DemandDetailing(_ref) {
  var children = _ref.children,
      reportTitle = _ref.reportTitle,
      reportKind = _ref.reportKind,
      detailingData = _ref.detailingData,
      rest = _objectWithoutPropertiesLoose(_ref, ["children", "reportTitle", "reportKind", "detailingData"]);

  return /*#__PURE__*/_react["default"].createElement(StyledDemandDetailing, _extends({}, rest, {
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 74,
      columnNumber: 5
    }
  }), /*#__PURE__*/_react["default"].createElement(_Fieldset["default"], {
    legend: reportTitle,
    bordered: true,
    pt: 4,
    px: ['30px', '40px', '40px', '40px'],
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 75,
      columnNumber: 7
    }
  }, /*#__PURE__*/_react["default"].createElement(ContentWrapper, {
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 76,
      columnNumber: 9
    }
  }, (detailingData || []).sort(function (first, last) {
    return last.createdAt - first.createdAt;
  }).map(function (item, index) {
    return /*#__PURE__*/_react["default"].createElement(TopicItemWrapper, {
      key: item.createdAt,
      className: getTopicClass(index, (detailingData || {}).length),
      __self: _this,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 80,
        columnNumber: 15
      }
    }, index === 1 && /*#__PURE__*/_react["default"].createElement(ReportBodyTitle, {
      __self: _this,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 81,
        columnNumber: 33
      }
    }, reportKind), /*#__PURE__*/_react["default"].createElement(_reflexbox.Flex, {
      alignItems: "center",
      justifyContent: "flex-start",
      __self: _this,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 82,
        columnNumber: 17
      }
    }, /*#__PURE__*/_react["default"].createElement(_UserAvatar["default"], {
      userName: item.interactionResponsibleName,
      displayName: item.interactionResponsibleName,
      mr: 2,
      __self: _this,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 83,
        columnNumber: 19
      }
    }), /*#__PURE__*/_react["default"].createElement(_Typography["default"], {
      fontSize: 1,
      color: "lightGrey",
      m: 0,
      __self: _this,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 84,
        columnNumber: 19
      }
    }, (0, _utils.formatDate)(item.createdAt))), /*#__PURE__*/_react["default"].createElement(TopicItem, {
      __self: _this,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 88,
        columnNumber: 17
      }
    }, /*#__PURE__*/_react["default"].createElement(_reflexbox.Box, {
      __self: _this,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 89,
        columnNumber: 19
      }
    }, /*#__PURE__*/_react["default"].createElement(_Typography["default"], {
      fontSize: 1,
      color: "#aaa",
      __self: _this,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 90,
        columnNumber: 21
      }
    }, item.comment)), index === 0 && /*#__PURE__*/_react["default"].createElement(_react["default"].Fragment, null, children), (item.documents || {}).length > 0 && item.documents.sort(function (first, last) {
      return last.createdAt - first.createdAt;
    }) && /*#__PURE__*/_react["default"].createElement(_reflexbox.Flex, {
      flexDirection: "column",
      justifyContent: "flex-start",
      alignItems: "flex-start",
      __self: _this,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 96,
        columnNumber: 21
      }
    }, /*#__PURE__*/_react["default"].createElement(_Typography["default"], {
      fontSize: 1,
      color: "lightGrey",
      fontWeight: 'bold',
      __self: _this,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 97,
        columnNumber: 23
      }
    }, "Anexos:"), item.documents.map(function (file) {
      return /*#__PURE__*/_react["default"].createElement(_Button["default"], {
        key: file.fileURL,
        as: "a",
        target: "_blank",
        rel: "noopener noreferrer",
        href: file.fileURL,
        linkbutton: true,
        color: "primary",
        linkButton: true,
        __self: _this,
        __source: {
          fileName: _jsxFileName,
          lineNumber: 101,
          columnNumber: 25
        }
      }, file.name);
    }))));
  }))));
};

var _default = DemandDetailing;
exports["default"] = _default;