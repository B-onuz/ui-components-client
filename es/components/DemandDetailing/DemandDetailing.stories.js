'use strict'

Object.defineProperty(exports, '__esModule', {
  value: true,
})
exports.DemandDetailingWithChild = exports.DemandDetailingExample = exports['default'] = void 0

var _react = _interopRequireDefault(require('react'))

var _DemandDetailing = _interopRequireDefault(require('./DemandDetailing'))

var _reflexbox = require('reflexbox')

var _Button = _interopRequireDefault(require('../Button'))

var _Typography = _interopRequireDefault(require('../Typography'))

var _this = void 0,
  _jsxFileName = '/home/junior/Fontes/ui-components-client/src/components/DemandDetailing/DemandDetailing.stories.js'

function _interopRequireDefault(obj) {
  return obj && obj.__esModule ? obj : { default: obj }
}

var _default = {
  title: 'DemandDetailing',
  component: _DemandDetailing['default'],
}
exports['default'] = _default
var data = [
  {
    interactionResponsibleName: 'Advogado Bonuz',
    isLawyer: true,
    createdAt: 1600977892300,
    comment:
      "Lorem ipsum, or lipsum as it is sometimes known, is dummy text used in laying out print, graphic or web designs. The passage is attributed to an unknown typesetter in the 15th century who is thought to have scrambled parts of Cicero's De Finibus Bonorum et Malorum for use in a type specimen book. It usually begins with:\n\n    \u201CLorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.\u201D\n    The purpose of lorem ipsum is to create a natural looking block of text (sentence, paragraph, page, etc.) that doesn't distract from the layout. A practice not without controversy, laying out pages with meaningless filler text can be very useful when the focus is meant to be on design, not content.\n\n    The passage experienced a surge in popularity during the 1960s when Letraset used it on their dry-transfer sheets, and again during the 90s as desktop publishers bundled the text with their software. Today it's seen all around the web; on templates, websites, and stock designs. Use our generator to get your own, or read on for the authoritative history of lorem ipsum.",
  },
  {
    interactionResponsibleName: 'Cliente',
    isLawyer: false,
    createdAt: 1600981780257,
    comment:
      'Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum ',
  },
  {
    interactionResponsibleName: 'Advogado Bonuz',
    isLawyer: true,
    createdAt: 1600981780267,
    comment:
      'Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum ',
    documents: [
      {
        fileURL: 'https://lorempixel.com/50/50/44',
        fileName: 'contrato_nda_v1.pdf',
        name: 'arquivo_final.png',
      },
      {
        fileURL: 'https://lorempixel.com/50/50',
        fileName: 'contrato_nda_v2.pdf',
        name: 'arquivo_semifinal.png',
      },
    ],
  },
]

var DemandDetailingExample = function DemandDetailingExample() {
  return /*#__PURE__*/ _react['default'].createElement(_DemandDetailing['default'], {
    reportTitle: 'Conclus\xE3o',
    reportKind: 'Hist\xF3rico',
    detailingData: data,
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 52,
      columnNumber: 45,
    },
  })
}

exports.DemandDetailingExample = DemandDetailingExample

var DemandDetailingWithChild = function DemandDetailingWithChild() {
  return /*#__PURE__*/ _react['default'].createElement(
    _DemandDetailing['default'],
    {
      reportTitle: 'Conclus\xE3o',
      reportKind: 'Hist\xF3rico',
      detailingData: data,
      __self: _this,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 55,
        columnNumber: 3,
      },
    },
    /*#__PURE__*/ _react['default'].createElement(
      _reflexbox.Flex,
      {
        justifyContent: 'space-between',
        my: 2,
        __self: _this,
        __source: {
          fileName: _jsxFileName,
          lineNumber: 56,
          columnNumber: 5,
        },
      },
      /*#__PURE__*/ _react['default'].createElement(
        _Typography['default'],
        {
          __self: _this,
          __source: {
            fileName: _jsxFileName,
            lineNumber: 57,
            columnNumber: 7,
          },
        },
        'Here comes any children content'
      ),
      /*#__PURE__*/ _react['default'].createElement(
        _reflexbox.Box,
        {
          __self: _this,
          __source: {
            fileName: _jsxFileName,
            lineNumber: 58,
            columnNumber: 7,
          },
        },
        /*#__PURE__*/ _react['default'].createElement(
          _Button['default'],
          {
            color: 'primary',
            m: 2,
            __self: _this,
            __source: {
              fileName: _jsxFileName,
              lineNumber: 59,
              columnNumber: 9,
            },
          },
          'Button 1'
        ),
        /*#__PURE__*/ _react['default'].createElement(
          _Button['default'],
          {
            outline: true,
            color: 'primary',
            m: 2,
            __self: _this,
            __source: {
              fileName: _jsxFileName,
              lineNumber: 62,
              columnNumber: 9,
            },
          },
          'Button 2'
        )
      )
    )
  )
}

exports.DemandDetailingWithChild = DemandDetailingWithChild
