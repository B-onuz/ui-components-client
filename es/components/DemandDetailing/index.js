"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _DemandDetailing = _interopRequireDefault(require("./DemandDetailing"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var _default = _DemandDetailing["default"];
exports["default"] = _default;