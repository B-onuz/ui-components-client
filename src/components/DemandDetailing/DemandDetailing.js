import React from 'react'
import styled from 'styled-components'
import { color, space } from 'styled-system'
import { Box, Flex } from 'reflexbox'
import Fieldset from '../Fieldset'
import UserAvatar from '../UserAvatar'
import Typography from '../Typography'
import Legend from '../Legend'
import Button from '../Button'
import { formatDate } from '../../utils'

const StyledDemandDetailing = styled.div`
  ${color}
  ${space}
  legend {
    font-size: 21px;
  }
  @media only screen and (max-width: 500px) {
    margin-left: 20px;
    padding-left: 0px;
  }
`

const ContentWrapper = styled.div`
  ${color}
  display: flex;
  flex-direction: column;
  border-left: 1px solid #dededf;
  margin-bottom: 40px;
`

const ReportBodyTitle = styled.h3`
  border-bottom: 1px solid #dededf;
  margin-bottom: 28px;
`

const TopicItemWrapper = styled.div`
  margin-bottom: 28px;
  margin-left: 38px;
  &.mainTopic {
    margin-left: -20px;
  }
  &.lastTopic {
    margin-bottom: 0;
  }

  @media only screen and (max-width: 500px) {
    margin-left: 10px;
  }
`

const TopicItem = styled.div`
  display: flex;
  flex-direction: column;
  padding-left: 52px;

  @media only screen and (max-width: 500px) {
    padding-left: 30px;
  }
`

const getTopicClass = (value, parameter) => {
  if (value === 0) {
    return 'mainTopic'
  } else if (value === parameter - 1) {
    return 'lastTopic'
  } else {
    return null
  }
}

const DemandDetailing = ({ children, reportTitle, reportKind, detailingData, ...rest }) => {
  return (
    <StyledDemandDetailing {...rest}>
      <Fieldset legend={reportTitle} bordered pt={4} px={['30px', '40px', '40px', '40px']}>
        <ContentWrapper>
          {(detailingData || [])
            .sort((first, last) => last.createdAt - first.createdAt)
            .map((item, index) => (
              <TopicItemWrapper key={item.createdAt} className={getTopicClass(index, (detailingData || {}).length)}>
                {index === 1 && <ReportBodyTitle>{reportKind}</ReportBodyTitle>}
                <Flex alignItems="center" justifyContent="flex-start">
                  <UserAvatar userName={item.interactionResponsibleName} displayName={item.interactionResponsibleName} mr={2} />
                  <Typography fontSize={1} color="lightGrey" m={0}>
                    {formatDate(item.createdAt)}
                  </Typography>
                </Flex>
                <TopicItem>
                  <Box>
                    <Typography fontSize={1} color="#aaa">
                      {item.comment}
                    </Typography>
                  </Box>
                  {index === 0 && <>{children}</>}
                  {(item.documents || {}).length > 0 && item.documents.sort((first, last) => last.createdAt - first.createdAt) && (
                    <Flex flexDirection="column" justifyContent="flex-start" alignItems="flex-start">
                      <Typography fontSize={1} color="lightGrey" fontWeight={'bold'}>
                        Anexos:
                      </Typography>
                      {item.documents.map((file) => (
                        <Button
                          key={file.fileURL}
                          as="a"
                          target="_blank"
                          rel="noopener noreferrer"
                          href={file.fileURL}
                          linkbutton
                          color="primary"
                          linkButton
                        >
                          {file.name}
                        </Button>
                      ))}
                    </Flex>
                  )}
                </TopicItem>
              </TopicItemWrapper>
            ))}
        </ContentWrapper>
      </Fieldset>
    </StyledDemandDetailing>
  )
}

export default DemandDetailing
