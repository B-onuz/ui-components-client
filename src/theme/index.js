const colors = {
  red: '#db2828',
  lightRed: '#ef4d5a',
  mediumRed: '#FF4F5D',
  darkRed: '#9c2543',
  fullDarkRed: '#550e33',
  orange: '#f2711c',
  yellow: '#fbbd08',
  olive: '#b5cc18',
  green: '#21BA45',
  teal: '#00b5ad',
  blue: '#2185d0',
  facebookBlue: '#3B5998',
  blueLink: '#4285F4',
  violet: '#6435c9',
  purple: '#a333c8',
  pink: '#e03997',
  brown: '#a5673f',
  grey: '#767676',
  textGrey: '#8D8D8D',
  lightGrey: '#6D6D6D',
  mediumGrey: '#838383',
  darkGrey: '#9A9A9A',
  boldGray: '#55555A',
  black: '#1b1c1d',
  white: '#fdfdfd',
  warning: '#7A4D05',
  success: '#1E561F',
  info: '#0E566C',
  danger: '#973937',
  cyan: '#1CB5AD',
  disabledGrey: '#ffffff1f',
  disabledGreyText: '#ffffff4d',
}

colors.primary = colors.lightRed
colors.secondary = colors.fullDarkRed
colors.default = '#e0e1e2'

const breakpoints = ['40em', '52em', '64em', '80em']

breakpoints.sm = breakpoints[0]
breakpoints.md = breakpoints[1]
breakpoints.lg = breakpoints[2]
breakpoints.xl = breakpoints[3]

const fontSizes = [12, 14, 16, 20, 24, 32]

export default {
  fontFamily: `'Lato', sans-serif`,
  colors,
  fontSizes,
  breakpoints,
}
