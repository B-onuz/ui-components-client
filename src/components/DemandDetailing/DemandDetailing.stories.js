import React from 'react'
import DemandDetailing from './DemandDetailing'
import { Box, Flex } from 'reflexbox'
import Button from '../Button'
import Typography from '../Typography'

export default {
  title: 'DemandDetailing',
  component: DemandDetailing,
}

const data = [
  {
    interactionResponsibleName: 'Advogado Bonuz',
    isLawyer: true,
    createdAt: 1600977892300,
    comment: `Lorem ipsum, or lipsum as it is sometimes known, is dummy text used in laying out print, graphic or web designs. The passage is attributed to an unknown typesetter in the 15th century who is thought to have scrambled parts of Cicero's De Finibus Bonorum et Malorum for use in a type specimen book. It usually begins with:

    “Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.”
    The purpose of lorem ipsum is to create a natural looking block of text (sentence, paragraph, page, etc.) that doesn't distract from the layout. A practice not without controversy, laying out pages with meaningless filler text can be very useful when the focus is meant to be on design, not content.

    The passage experienced a surge in popularity during the 1960s when Letraset used it on their dry-transfer sheets, and again during the 90s as desktop publishers bundled the text with their software. Today it's seen all around the web; on templates, websites, and stock designs. Use our generator to get your own, or read on for the authoritative history of lorem ipsum.`,
  },
  {
    interactionResponsibleName: 'Cliente',
    isLawyer: false,
    createdAt: 1600981780257,
    comment:
      'Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum ',
  },
  {
    interactionResponsibleName: 'Advogado Bonuz',
    isLawyer: true,
    createdAt: 1600981780267,
    comment:
      'Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum Lorem Ipsum ',
    documents: [
      {
        fileURL: 'https://lorempixel.com/50/50/44',
        fileName: 'contrato_nda_v1.pdf',
        name: 'arquivo_final.png',
      },
      {
        fileURL: 'https://lorempixel.com/50/50',
        fileName: 'contrato_nda_v2.pdf',
        name: 'arquivo_semifinal.png',
      },
    ],
  },
]

export const DemandDetailingExample = () => <DemandDetailing reportTitle="Conclusão" reportKind="Histórico" detailingData={data} />

export const DemandDetailingWithChild = () => (
  <DemandDetailing reportTitle="Conclusão" reportKind="Histórico" detailingData={data}>
    <Flex justifyContent="space-between" my={2}>
      <Typography>Here comes any children content</Typography>
      <Box>
        <Button color="primary" m={2}>
          Button 1
        </Button>
        <Button outline color="primary" m={2}>
          Button 2
        </Button>
      </Box>
    </Flex>
  </DemandDetailing>
)
