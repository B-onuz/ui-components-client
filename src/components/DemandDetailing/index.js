import DemandDetailing from './DemandDetailing'

export default DemandDetailing
