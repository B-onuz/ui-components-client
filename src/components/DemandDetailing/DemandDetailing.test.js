import React from 'react'
import DemandDetailing from './DemandDetailing.js'
import renderer from 'react-test-renderer'
import { ThemeProvider } from 'styled-components'
import theme from '../../theme'
import 'jest-styled-components'

test('Should render DemandDetailing', () => {
  const component = renderer.create(
    <ThemeProvider theme={theme}>
      <DemandDetailing />
    </ThemeProvider>
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
