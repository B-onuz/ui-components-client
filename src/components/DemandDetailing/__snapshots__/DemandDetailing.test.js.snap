// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render DemandDetailing 1`] = `
.c1 {
  padding: 0;
  margin: 0;
  border-radius: 4px;
  border: 1px solid #DEDEDF;
  padding-top: 32px;
  padding-left: 30px;
}

.c1 legend {
  margin: 1em;
  padding: .3em;
}

.c0 legend {
  font-size: 21px;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  border-left: 1px solid #dededf;
  margin-bottom: 40px;
}

@media screen and (min-width:40em) {
  .c1 {
    padding-left: 40px;
  }
}

@media screen and (min-width:52em) {
  .c1 {
    padding-left: 40px;
  }
}

@media screen and (min-width:64em) {
  .c1 {
    padding-left: 40px;
  }
}

@media only screen and (max-width:500px) {
  .c0 {
    margin-left: 20px;
    padding-left: 0px;
  }
}

<div
  className="c0"
>
  <fieldset
    className="c1"
  >
    <div
      className="c2"
    />
  </fieldset>
</div>
`;
